### PR TITLE
add multicluster design docs

### PR DESCRIPTION
**Pull request by zryfish**
Add multicluster design docs

### CODE REVIEW

File: developer-guide/concepts-and-designs/multicluster/architecture.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# Multiple Cluster Architecture
+
+## Introduction
+
+Today, it's very common for organizations to run a handful of Kubernetes Clusters. Each kubernetes cluster is a relatively self-contained unit. And there are several teams working on tying all clusters, such as kubefed. More and more KubeSphere users and customers have expressed their kneed interests in managing all clusters through KubeSphere.
```
Comment (rayzhou2017) [inline developer-guide/concepts-and-designs/multicluster/architecture.md:5]: kneed interests? Is kneed a good word here?

File: developer-guide/concepts-and-designs/multicluster/architecture.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+## Summary
+For KubeSphere users and customers, the two highest priority ones are relate to Operating clusters in one place and High Availability.
```
Comment (rayzhou2017) [inline developer-guide/concepts-and-designs/multicluster/architecture.md:8]: the two highest priority ones are relate to Operating clusters in one place and High Availability. => the two highest priority ones are operating clusters in one place and providing high availability at the same time.

File: developer-guide/concepts-and-designs/multicluster/architecture.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+## What is KubeSphere multiple cluster?
+KubeSphere multiple cluster is the ability to operate clusters in one place, which is KubeSphere, and extend kubefed to achieve High Availability.
+
+## Architecture
+The overall architecture of a control plane is shown as following.
```
Comment (rayzhou2017) [inline developer-guide/concepts-and-designs/multicluster/architecture.md:14]: as follows or as the following

File: developer-guide/concepts-and-designs/multicluster/architecture.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+![](architecture.png)
+
+Some design principles we are following in this architecture.
+
+1. Keep the underlying KubeSphere clusters independent. The should have no knowledge of control plane or of each other.
```
Comment (rayzhou2017) [inline developer-guide/concepts-and-designs/multicluster/architecture.md:20]: The should have no knowledge => They should have no knowledg

File: developer-guide/concepts-and-designs/multicluster/architecture.md (hunk 1 above)
Comment (zryfish) [reply developer-guide/concepts-and-designs/multicluster/architecture.md:5]: typo, kneen

File: developer-guide/concepts-and-designs/multicluster/architecture.md (hunk 2 above)
Comment (zryfish) [reply developer-guide/concepts-and-designs/multicluster/architecture.md]: changed as suggested

File: developer-guide/concepts-and-designs/multicluster/architecture.md (hunk 3 above)
Comment (zryfish) [reply developer-guide/concepts-and-designs/multicluster/architecture.md]: fixed

File: developer-guide/concepts-and-designs/multicluster/architecture.md (hunk 4 above)
Comment (zryfish) [reply developer-guide/concepts-and-designs/multicluster/architecture.md]: fixed

**Pull request merged**